doc/specs/stdlib_string_type.md

```diff
@@ -0,0 +1,1607 @@
+---
+title: string type
+---
+
+# The `stdlib_string_type` module
+
+[TOC]
+
+## Introduction
+
+The `stdlib_string_type` provides a derived type holding an arbitrary sequence
+of characters compatible with most Fortran intrinsic character procedures as
+well as operators for working with character variables and constants.
+
+
+## Derived types provided
+
+
+<!-- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -->
+### The `string_type` derived type
+
+The `string_type` is defined as a non-extenable derived type representing a
+sequence of characters. The internal representation of the character sequence
+is implementation dependent and not visible for the user of the module.
+
+#### Status
+
+Experimental
+
+
+## Procedures and methods provided
+
+Procedures returning `string_type` instances can usually used in elemental
+context, while procedures returning scalar character values can only be
+used in a pure way.
+
+
+<!-- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -->
+### Constructor for empty string
+
+The module defines a default constructor to create an empty string type.
+
+#### Description
+
+Creates a string instance representing an empty string.
+
+#### Syntax
+
+`res = [[stdlib_string_type(module):string_type(interface)]] ()`
+
+#### Status
+
+Experimental
+
+#### Class
+
+Elemental function.
+
+#### Argument
+
+None.
+
+#### Result value
+
+The result is an instance of `string_type`.
```

Comment:
```suggestion
The result is an instance of `string_type` with a lenght equal to 0.
```

Reply:
Watch out for "length" in the proposed correction.
```suggestion
The result is an instance of `string_type` with zero length.
```